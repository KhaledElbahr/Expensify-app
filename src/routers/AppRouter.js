import React from 'react';
import { BrowserRouter as Router, Routes, Route, NavLink } from 'react-router-dom';
import ExpenseList from './../Expensify/ExpenseList';
import ExpenseListFilters from './../Expensify/ExpenseListFilters';
import AddExpensePage from './../Expensify/AddExpensePage';
import { EditExpensePage } from './../Expensify/EditExpensepage';

export const Navbar = () => (
    <header>
        <h1>Expensify</h1>
            <div>
                <NavLink to="/" className="link active" exact="true">Dashboard</NavLink>
                <NavLink to="/add-expense" className="link active">Create Expense</NavLink>
                <NavLink to="/about" className="link active">About</NavLink>
                <NavLink to="/contact-us" className="link active">Contact Us</NavLink>
            </div>
    </header>
)

export const HomePage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
)

const AboutPage = () => (
    <div>
        About Page works successfully!
    </div>
)

const ContactUsPage = () => (
    <div>
        Contact us Page works successfully!
    </div>
)

export const NotFoundPage = () => (
    <div style={{ textAlign: 'center' }}>
        <h1>Not Found Page</h1>
        <h2>404</h2>
        <NavLink to="/" replace>Go back to Home Page -></NavLink>
    </div>
)
  
const AppRouter = () => (
    <Router>
        <div>
            <Navbar />
            <Routes>
                <Route path="/" element={<HomePage />} exact={true} />
                <Route path="/add-expense" element={<AddExpensePage />} />
                <Route path="/edit-expense/:id" element={<EditExpensePage />} />
                <Route path="/about" element={<AboutPage />} />
                <Route path="/contact-us" element={<ContactUsPage />} />
                <Route path="*" element={<NotFoundPage />} />
            </Routes>
        </div>
    </Router>
);

export default AppRouter;