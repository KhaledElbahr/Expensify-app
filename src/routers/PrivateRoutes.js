import React from 'react';
import { connect } from 'react-redux';
import { Outlet, Navigate } from 'react-router-dom';
import Header from '../Expensify/Header';

export const PrivateRoutes = ({ isAuthenticated }) => (
    isAuthenticated ? (<> <Header /> <Outlet /> </>) : <Navigate to="/" />
);

const mapStateToProps = (state) => ({
    isAuthenticated: !!state.auth.uid
});

export default connect(mapStateToProps)(PrivateRoutes);
