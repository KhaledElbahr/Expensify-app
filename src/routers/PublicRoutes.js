import React from 'react';
import { connect } from 'react-redux';
import { Outlet, Navigate } from 'react-router-dom';
import Header from '../Expensify/Header';

export const PublicRoutes = ({ isAuthenticated }) => (
    isAuthenticated ? <Navigate to="/dashboard" /> : <Outlet />
);

const mapStateToProps = (state) => ({
    isAuthenticated: !!state.auth.uid
});

export default connect(mapStateToProps)(PublicRoutes);
