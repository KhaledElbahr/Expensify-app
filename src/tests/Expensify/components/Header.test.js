import React from 'react';
// import renderer from 'react-test-renderer';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { Navbar } from './../../../routers/AppRouter';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

test('should render Header correctly', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper.find('h1').text()).toBe('Expensify');
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
});

// test('should render Header correctly', () => {
//     const renderer = new ReactShallowRenderer();
//     renderer.render(<Navbar />);
//     const result = renderer.getRenderOutput();
//     console.log(result.props.children);

//     expect(result).toMatchSnapshot();
//     expect(result.type).toBe('header');
// });