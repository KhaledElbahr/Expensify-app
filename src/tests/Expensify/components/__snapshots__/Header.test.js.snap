// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <div>
    <NavLink
      className="link active"
      exact="true"
      to="/"
    >
      Dashboard
    </NavLink>
    <NavLink
      className="link active"
      to="/add-expense"
    >
      Create Expense
    </NavLink>
    <NavLink
      className="link active"
      to="/about"
    >
      About
    </NavLink>
    <NavLink
      className="link active"
      to="/contact-us"
    >
      Contact Us
    </NavLink>
  </div>
</header>
`;
